### PR TITLE
feat(messenger): print IP address in string format

### PR DESCRIPTION
**Pull request by handong32**
adding ToString() will be useful for printing
IP address to screen

### CODE REVIEW

File: baremetal/src/include/ebbrt/Messenger.h [hunk 1]
```diff
@@ -47,6 +47,33 @@ class Messenger : public StaticSharedEbb<Messenger>, public CacheAligned {
       return std::string(reinterpret_cast<const char*>(ip.toArray().data()), 4);
     }
 
+    std::string ToString() {
+      auto data = ip.toArray().data();
+      if (str.size() != 4)
+        throw std::runtime_error("ebbrt::Messsenger::NetworkId.ToString() - "
+                                 "NetworkId size not equal to 4");
```
Comment (dschatzberg) [inline baremetal/src/include/ebbrt/Messenger.h]: use an assert here

File: baremetal/src/include/ebbrt/Messenger.h [hunk 2]
```diff
@@ -47,6 +47,33 @@ class Messenger : public StaticSharedEbb<Messenger>, public CacheAligned {
...
+      ret.append(".");
+
+      ss << data[2];
+      ret.append(ss.str());
+
+      ret.append(".");
+
+      ss << data[3];
+      ret.append(ss.str());
+      return ret;
```
Comment (dschatzberg) [inline baremetal/src/include/ebbrt/Messenger.h]: use std::to_string for each of the types and the + operator

**Comment by jmcadden** [comment]
String conversion should be implemented as a part of the Ipv4Address type.  e.g., `ip.to_string()`

See how Messanger::NetworkId::ToString defined on the hosted side: https://github.com/SESA/EbbRT/blob/master/hosted/src/include/ebbrt/Messenger.h#L42

File: baremetal/src/include/ebbrt/Messenger.h [hunk 3]
```diff
@@ -47,6 +47,33 @@ class Messenger : public StaticSharedEbb<Messenger>, public CacheAligned {
...
+    }
+
    private:
     Ipv4Address ip;
```
Comment (jmcadden) [inline baremetal/src/include/ebbrt/Messenger.h]: private member variables should end with an underscore (e.g., ip_) - I know this isn't new code but let's fix it anyway.

**Comment by jmcadden** [comment]
any update?

**Comment by dschatzberg** [comment]
Closing this as it appears to have been abandoned

**Pull request closed without merging**